### PR TITLE
More preference datasets

### PR DESCRIPTION
**Pull request by natolambert**
* Many HelpSteer2 configs
* A few Nectar configurations

### CODE REVIEW

File: docs/data/preference-data.md [hunk 1]
```diff
@@ -21,9 +21,38 @@ python scripts/data/preferences/hh-helpful.py --push_to_hub --hf_entity=ai2-adap
...
+python scripts/data/preferences/helpsteer2.py --push_to_hub --hf_entity=ai2-adapt-dev --aspects_to_ignore verbosity
+```
+The binarization weighting that Nvidia recommends can be used with:
+```
+python scripts/data/preferences/helpsteer2_nvidia.py --push_to_hub --hf_entity ai2-adapt-dev
+```
+Some examples include:
+* [ai2-adapt-dev/helpsteer-2-binarized-above-2.0-margin-0.5-ignore-verbosity](https://huggingface.co/datasets/ai2-adapt-dev/helpsteer-2-binarized-above-2.0-margin-0.5-ignore-verbosity)
+* [ai2-adapt-dev/helpsteer-2-binarized-ignore-verbosity](https://huggingface.co/datasets/ai2-adapt-dev/helpsteer-2-binarized-ignore-verbosity)
+* [ai2-adapt-dev/helpsteer2-binarized-nvidia-spec](https://huggingface.co/datasets/ai2-adapt-dev/helpsteer2-binarized-nvidia-spec)
```
Comment (vwxyzjn) [inline docs/data/preference-data.md]: What is the difference between these? Could we do something like below, basically put the generated dataset and command side by side. 

```
python scripts/data/preferences/helpsteer2.py --push_to_hub --min_score 2.5 --hf_entity=ai2-adapt-dev
# created https://huggingface.co/datasets/ai2-adapt-dev/helpsteer-2-binarized-above-2.0-margin-0.5-ignore-verbosity
python scripts/data/preferences/helpsteer2.py --push_to_hub --aspects_to_ignore verbosity  --hf_entity=ai2-adapt-dev
# created https://huggingface.co/datasets/ai2-adapt-dev/helpsteer-2-binarized-ignore-verbosity

File: scripts/data/preferences/helpsteer2.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Binarize HelpSteer 2 dataset and optionally upload to Hugging Face Hub."
+    )
+    parser.add_argument(
+        "--aspects_to_ignore",
+        nargs="*",
+        default=[],
+        help="Aspects to ignore during binarization (default: none)"
+    )
```
Comment (vwxyzjn) [inline scripts/data/preferences/helpsteer2.py]: Very nice feature!

File: scripts/data/preferences/nectar.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    if deduplication:
+        print("Deduplication enabled. Loading UltraFeedback dataset...")
+        ultra_feedback = load_dataset("openbmb/UltraFeedback", split="train")
+        print(f"UltraFeedback dataset size: {len(ultra_feedback)}")
+        
+        # Create a set of UltraFeedback instructions for faster lookup
+        ultra_feedback_instructions = set(ultra_feedback['instruction'])
+
+        # Filter Nectar dataset, excluding lmsys-chat-1m and anthropic-hh
+        filtered_dataset = nectar_dataset.filter(lambda example: example['source'][0] not in ["lmsys-chat-1m", "anthropic-hh"])
```
Comment (vwxyzjn) [inline scripts/data/preferences/nectar.py]: Why is this called `deduplication`? Would it be more appropriate to call it `--remove-sources "lmsys-chat-1m" "anthropic-hh"`?

**Review by vwxyzjn — APPROVED** [review]
Very good addition to the dataset @natolambert :) Some minor comments!

File: scripts/data/preferences/nectar.py (hunk 3 above)
Comment (natolambert) [reply scripts/data/preferences/nectar.py]: It's for deduplication of prompts found in UltraFeedback, our other popular dataset. There's no overlap with LMSYS + HH so we handle them differently.

File: docs/data/preference-data.md (hunk 1 above)
Comment (natolambert) [reply docs/data/preference-data.md]: I'm not sure its worth the effort. I don't know if 1 person has used this readme in actuality. As people use it more, they can document more.

**Pull request merged**